FreeRTOS/FreeRTOS-Plus-TCP
# Add NXP RT1060 network interface

Add NXP RT1060 network interface back

Description
-----------

The network interface of NXP RT1060 had been removed temporarily because it's out of date when IPv6 merging(#953).
In this PR, we update the interface and add it back to portable layer with supporting multi-interfaces/endpoints.

Test Steps
-----------

Thank @AniruddhaKanhere for providing a project to test.
1. Run echo server on device on [FreeRTOSPlusTCPDemo](https://github.com/ActoryOu/iot-reference-nxp-rt1060/tree/FreeRTOSPlusTCPDemo) branch. This project enables backwards compatible on FreeRTOS-Plus-TCP.
    1. Follow steps [GSG](https://github.com/ActoryOu/iot-reference-nxp-rt1060/blob/FreeRTOSPlusTCPDemo/GSG.md) to set up the environments, including [hardware setup](https://github.com/ActoryOu/iot-reference-nxp-rt1060/blob/FreeRTOSPlusTCPDemo/GSG.md#21-setting-up-device), [importing software project](https://github.com/ActoryOu/iot-reference-nxp-rt1060/blob/FreeRTOSPlusTCPDemo/GSG.md#22-importing-and-building-projects), and [bootloader](https://github.com/ActoryOu/iot-reference-nxp-rt1060/blob/FreeRTOSPlusTCPDemo/GSG.md#3-prepare-and-run-the-bootloader).
    2. We use aws_iot_defender project to create a echo server locally. So please compile and run aws_iot_defender.
    3. Use another device on same LAN as echo client to test the echo server on device.
1. Run echo server on device on [FreeRTOSPlusTCPDemoNoBackwardCompatible](https://github.com/ActoryOu/iot-reference-nxp-rt1060/tree/FreeRTOSPlusTCPDemoNoBackwardCompatible) branch. This project disables backwards compatible on FreeRTOS-Plus-TCP.
    1. Follow steps [GSG](https://github.com/ActoryOu/iot-reference-nxp-rt1060/blob/FreeRTOSPlusTCPDemo/GSG.md) to set up the environments, including [hardware setup](https://github.com/ActoryOu/iot-reference-nxp-rt1060/blob/FreeRTOSPlusTCPDemo/GSG.md#21-setting-up-device), [importing software project](https://github.com/ActoryOu/iot-reference-nxp-rt1060/blob/FreeRTOSPlusTCPDemo/GSG.md#22-importing-and-building-projects), and [bootloader](https://github.com/ActoryOu/iot-reference-nxp-rt1060/blob/FreeRTOSPlusTCPDemo/GSG.md#3-prepare-and-run-the-bootloader).
    2. We use aws_iot_defender project to create a echo server locally. So please compile and run aws_iot_defender.
    3. Use another device on same LAN as echo client to test the echo server on device.

Note that only IPv4 are verified with/without backward compatible configuration.

Checklist:
----------



Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
